sklearn/preprocessing/_data.py

```diff
@@ -1713,9 +1718,10 @@ def robust_scale(
         .. versionadded:: 0.18
 
     copy : bool, default=True
-        Set to `False` to perform inplace row normalization and avoid a
-        copy (if the input is already a numpy array or a scipy.sparse
-        CSR matrix and if axis is 1).
+        If False, try to avoid a copy and scale in place.
+        This is not guaranteed to always work in place; e.g. if the data is
+        not a NumPy array, or is a scipy.sparse CSR matrix, or is not of dtype
+        float, a copy may still be returned.
```

Comment:
Honestly I am started to think that aiming to be fully accurate in the docstring about the copy=False behaviour is not going to be workable. I would instead give 1 or 2 simple examples of case where data is copied even with `copy=False`.

For example in this case I would say something like "e.g. if the data is a Numpy array with a dtype not float, a copy will be returned". Basically let's only look at the first check_array call and give an example where even with `copy=False` it will return a copy.

In this particular case the first `check_array` call accept csr/csc so no copy but then it depends on the `RobustScaler`, for example the CSR will be turned into CSC in `RobustScaler.fit` which may share some indices (so partial copy?) ...

Reply:
There are two ways of formulating the docstring. 1) We present the cases where `copy = False` works exhaustively 2) we present examples where `copy=False` fails. I completely agree that the second approach is more feasible. 

As you mention, copies probably happen in different functions called by (in this case) maxabs_scale and simply looking at the check_array arguments is misleading. Is however the source of the copy relevant for the docstring? 

I see two ways moving forward. 

1) I create some sort of test that confirms copies are made in case `x`. For example, I've created a colab where I already check that a copy is made in all the cases that I've listed (even if these don't happen because of check_array, I'm not tracking the source). I can adapt it based on a proposed template. 

2) I include in all dosctrings only the example `data is not a NumPy array` which triggers a copy through `check_array` in all the functions that are part of this PR. (This is basically also what you are proposing if I understand correctly)

Reply:
It feels like we are on the same line, i.e. giving one or two simple counter-examples, where copy=False still creates a copy so 2).

About ways for moving forward, I would say the testing approach seems too much work and not worth it.

So I would say:
- use the simple example of the numpy array with a non-float dtype everywhere we can, i.e. the first `check_array` call has `dtype=FLOAT_DTYPES`. This was the confusion in the original issue. More generally I am finding only now (sorry for realising this late) that "is not a Numpy array" is a bit unclear because a scipy sparse matrix is not a numpy array and yet in some cases scipy sparse matrix will not be copied.
- treat the remaining cases with more care (I don't have a good view how many there are)

> As you mention, copies probably happen in different functions called by (in this case) maxabs_scale and simply looking at the check_array arguments is misleading

Just for completeness, I was trying to say that looking at the first `check_array` is enough to find a counter-example where copy=False still creates a copy.

Reply:
Also I would use the wording and use "a copy will be returned" instead of "a copy may be returned" for the counter-example we give.

In some way it would be nice to find a way to convey that this is a example where copy=False still returns a copy but that there may be (plenty of) others.